### PR TITLE
#558 - Added basic user preferences. Next - need to add validation fo…

### PR DESCRIPTION
**Pull request by aviklai**
…r the inputs and design the settings modal.

### CODE REVIEW

File: models.py [hunk 1]
```diff
@@ -529,6 +529,43 @@ def is_anonymous(self):
     def get_id(self):
         return self.id
 
+class GeneralPreferences(Base):
```
Comment (danielhers) [inline models.py]: Please add an `alembic` migration to create these tables.

File: static/js/preferencesdialog.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    close: function () {
+        this.modal.modal("hide");
+    },
+    submit: function () {        
+        var lat = this.$el.find("#prefLat").val();
+        var lon = this.$el.find("#prefLon").val();
+        var distance = this.$el.find("#prefDistance").val();
+        var accident_severity = this.$el.find("#prefAccidentSeverity :selected").val();
+        var history_report = this.$el.find("#prefHistoryReportTrue")
+        if ($(history_report).is(":checked")) {
```
Comment (danielhers) [inline static/js/preferencesdialog.js]: Why not `history_report = $(history_report).is(":checked")`?

**Comment by danielhers** [comment]
Looks good, but I want to see it running. Unfortunately I can't seem to log in on https://anyway-unstable-pr-562.herokuapp.com even though I added it to the allowed URLs on Facebook. Any idea why?

File: models.py (hunk 1 above)
Comment (aviklai) [inline models.py]: Done.

**Comment by aviklai** [comment]
@danielhers You can try to login now. I sorted out the allowed url for the unstable version.

File: alembic/versions/2438f7d83478_add_user_preferences.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+def upgrade():
+    op.create_table(
+        'general_preferences',
+        sa.Column('user_id', sa.Integer, ForeignKey('users.id'), primary_key=True),
+        sa.Column('minimum_displayed_severity', sa.Integer, nullable=True),
+        sa.Column('resource_type', sa.String(64), nullable=True),
+    )	
+	
+	op.create_table(
```
Comment (danielhers) [inline alembic/versions/2438f7d83478_add_user_preferences.py]: Fix the indent here. It's a syntax error.

**Comment by aviklai** [comment]
@danielhers I Fixed the syntax error in the alembic migration.

File: alembic/versions/2438f7d83478_add_user_preferences.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+depends_on = None
+
+from alembic import op
+import sqlalchemy as sa
+
+
+def upgrade():
+    op.create_table(
+        'general_preferences',
+        sa.Column('user_id', sa.Integer, ForeignKey('users.id'), primary_key=True),
```
Comment (danielhers) [inline alembic/versions/2438f7d83478_add_user_preferences.py]: `ForeignKey` should have `sa.` before

**Comment by aviklai** [comment]
@danielhers Fixed foreign key constraint in alembic migration file.

**Comment by danielhers** [comment]
Now the UX should be improved, since it's quite bad...

**Comment by aviklai** [comment]
I know. @simonaho - we have a UX person - right? Can we assign to her the improvement of all of those views?

**Pull request merged**